src/app/app.component.ts

```diff
@@ -133,11 +147,37 @@ export class AppComponent implements OnInit {
     });
   }
 
+  isMobile() {
+    let isMobileDevice = false;
+    if ('maxTouchPoints' in navigator as any) {
+      isMobileDevice = 0 < navigator.maxTouchPoints;
+    } else if ('msMaxTouchPoints' in navigator as any) {
+      isMobileDevice = 0 < navigator.msMaxTouchPoints;
+    } else {
+      const mQ = window.matchMedia?.('(pointer:coarse)');
+      if (mQ?.media === '(pointer:coarse)') {
+        isMobileDevice = !!mQ.matches;
+      } else if ('orientation' in window) {
+        isMobileDevice = true;
+      } else {
+        const UA = navigator.userAgent;
+        isMobileDevice = (
+          /\b(BlackBerry|webOS|iPhone|IEMobile)\b/i.test(UA) ||
+          /\b(Android|Windows Phone|iPad|iPod)\b/i.test(UA)
+        );
```

Comment:
Probably a good sign to make use of an NPM package (such as [bowser](https://www.npmjs.com/package/bowser) or [ua-parser-js](https://www.npmjs.com/package/ua-parser-js)) instead of writing some regex magic.

Reply:
Main Cases should be covered by the touchscreen feature detection + media query.. this is really the last resort which most likely is not really required as all major modern browsers and devices should fullfill the previous conditions. Therefore i didn't wanted to bloat up the bundle more (also anways not that accurate).. up to you - i can add it if you like